src/daemon.py

```diff
@@ -81,12 +85,55 @@ def __search_task(self, job_id: JobId) -> None:
         self.db.update_job_files(job_id, file_count)
         self.db.agent_start_job(self.group_id, job_id, iterator)
 
+    def __load_plugins(self) -> None:
+        self.plugin_config_version = self.db.get_plugin_config_version()
+        active_plugins = []
+        for plugin_class in METADATA_PLUGINS:
+            plugin_name = plugin_class.get_name()
+            plugin_config = self.db.get_plugin_configuration(plugin_name)
+            try:
+                active_plugins.append(plugin_class(self.db, plugin_config))
+                logging.info("Loaded %s plugin", plugin_name)
+            except Exception:
+                logging.exception("Failed to load %s plugin", plugin_name)
+        self.active_plugins = active_plugins
+
+    def __initialize_agent(self) -> None:
+        self.__load_plugins()
+        plugins_spec = {
+            plugin_class.get_name(): plugin_class.config_fields
+            for plugin_class in METADATA_PLUGINS
+        }
+        self.db.register_active_agent(
+            self.group_id,
+            self.ursa_url,
+            plugins_spec,
+            [
+                active_plugin.get_name()
+                for active_plugin in self.active_plugins
+            ],
+        )
+
     def __update_metadata(
         self, job: JobId, file_path: str, matches: List[str]
     ) -> None:
         """ After finding a match, push it into a database and
         update the related metadata """
-        match = MatchInfo(file_path, {}, matches)
+        if self.db.get_plugin_config_version() != self.plugin_config_version:
```

Comment:
Ugh, it's not a good place to do this - __update_metadata is in a realatively tight loop.

Move to a beginning of `__execute_yara` - in a worst case changes will be applied few seconds later, but we'll save a lot of requests to redis.

Reply:
(or to a beginning of `__yara_task`, that's good too)

Reply:
Moved to the beginning of `__execute_yara`